crates/autopilot/src/run_loop.rs

```diff
@@ -513,6 +517,28 @@ pub fn solve_request(
                             .collect()
                     };
                 let order_is_untouched = remaining_order.executed_amount.is_zero();
+                let fee_policy = match (
```

Comment:
This block of code will explode in complexity in the next PRs and is conveniently embedded in an already very long method. I think we should create a "domain" style fee policy type which is instantiated from the fee_policy arguments and exposes a method `fn apply(&self, &mut order)`. This allows us to encapsulate the logic of
1. How to parse our variety of different arguement combinations into a coherent fee policy
2. How to apply a concrete fee policy to a concrete order.

Reply:
refactored the arguments into enum, and then created a "to_domain" function for conversion, similar to what we have in `driver`. It should be a step forward to ddd in autopilot.